### ISSUE TITLE
Quickstart instructions not accurate?

### ISSUE THREAD

**Issue by helliger** [projectblacklight/blacklight#1393]
Has anyone gone through the quickstart instructions to verify they are still accurate? 
https://github.com/projectblacklight/blacklight/wiki/Quickstart

Following the instructions using both the easy way and hard way results in an installation lacking a blacklight_core. 

After creating a core manually and making some adjusts to the schema I was able to index some marc records however they are not searchable. I suspect there's either a missing step or an outdated step.

I saw a report of a  problem with the instructions resulting in a hydra install but there was no resolution. Just a possible workaround. 

Thanks

**Comment by natehill** [comment]
I had the same problem.  
Spent a couple of hours trying both the easy way and the hard way and get:
There exists no core with name "blacklight-core"

**Comment by jkeck** [comment]
This actually looks like an issue with the installer.

The instance of solr that is being installed is `hydra-jetty` and not `blacklight-jetty`.

I have submitted a pull-request to address this, and if accepted, will come out with the next release.

Apologies for the inconvenience, and thank you very much for bringing this to our attention.

**Comment by helliger** [comment]
Thanks. The workaround I did is outlined below. Do any gotcha's jump out? It installed a blacklight-core and appears to run as expected although I know very little about a standard Blacklight install.



1. add solr_wrapper to the gemfile

2. added the following to the bottom of the Rakefile

require 'solr_wrapper/rake_task'
ZIP_URL = "https://github.com/projectblacklight/blacklight-jetty/archive/v4.10.4.zip"


3. Ran the Blacklight install with --solr_wrapper added to the command line.

rails generate blacklight:install --devise --marc --jettywrapper --solr_wrapper

**Comment by natehill** [comment]
thanks @jkeck - but when I follow the workaround @helliger presents, I cannot start jetty:

▶ rake jetty:start
rake aborted!
Don't know how to build task 'jetty:start' (see --tasks)
/Users/natehill/.rvm/gems/ruby-2.1.1/bin/ruby_executable_hooks:15:in `eval'
/Users/natehill/.rvm/gems/ruby-2.1.1/bin/ruby_executable_hooks:15:in `<main>'

**Comment by jkeck** [comment]
I'm not entirely sure where the app was through the install process in the comment above.  

One thing you may want to try is to follow the [Creating a new application the hard way](https://github.com/projectblacklight/blacklight/wiki/Quickstart#creating-a-new-application-the-hard-way) quickstart guide.  Before you run the `blacklight:install` rails generator do the following in your application.

1) Add `gem "jettywrapper", ">= 2.0"` to your Gemfile
2) Add `gem "rsolr", "~> 1.0"` to your Gemfile
3) Copy the `jetty.yml` file located at https://github.com/projectblacklight/blacklight/blob/master/config/jetty.yml to your application's `config` directory.
4) Add the following to your `Rakefile`:
```ruby
ZIP_URL = 'https://github.com/projectblacklight/blacklight-jetty/archive/v4.10.4.zip'
require 'jettywrapper'
```
5) run `bundle install`
6) Continue on with the instructions starting by running `rails generate blacklight:install --devise --marc --jettywrapper`

That's totally untested, so hopefully it works.  This is effectively manually running all of the operations that are happening in the [Solr4Generator](https://github.com/projectblacklight/blacklight/blob/master/lib/generators/blacklight/solr4_generator.rb), which I believe is not running during the current install.

**Comment by jkeck** [comment]
And I should add, that the goal is to have this all happen for you automatically, and this is just a workaround until we can cut a release w/ the install process fixed.

**Comment by natehill** [comment]
Much thanks @jkeck this all worked after I made two small changes: 

1. the jetty.yml file at https://github.com/projectblacklight/blacklight/blob/master/config/jetty.yml specifies java 1.7 and I'm running 1.8

2. adding `gem "rsolr", "~> 1.0"` to the Gemfile created a version conflict, so I didn't do it.

Appreciate your help.

**Comment by yinlinchen** [comment]
This is what I did. 
I modify the template.demo.rb file, update blacklight to 6.1, remove --marc  the file become this https://github.com/yinlinchen/blacklight/blob/master/template.demo.rb
After that, use these commands and a demo site is up and run. 
$ mkdir projects
$ cd projects
$ rails new search_app -m https://raw.githubusercontent.com/yinlinchen/blacklight/master/template.demo.rb
$ cd search_app
#Add this line to Gemfile:
gem 'blacklight-marc'
$ bundle
# Copy blacklight.scss to blacklight.css.scss in app/assets/stylesheets/
$ cd app/assets/stylesheets/
$ cp blacklight.scss blacklight.css.scss
$ rails generate blacklight:marc:install
$ rake solr:marc:index_test_data
$ rails server

ps, There is a [PR](https://github.com/projectblacklight/blacklight-marc/pull/33) for blacklight.scss but not yet merged.

**Comment by Yongyao** [comment]
Hi guys,

I am an Elasticsarch and Java guy, and know a bit solr, but nothing at all about Ruby. I am just trying to evaluate Blacklight by going through the quickstart. When I got to 
`rake solr:marc:index_test_data`, it gives me the error below.
```
2016-09-05T11:51:25-04:00  INFO    Traject::SolrJsonWriter writing to 'http://127.0.0.1:8983/solr/blacklight-core/update' in batches of 100 with 1 bg threads
2016-09-05T11:51:25-04:00  INFO    Indexer with 1 processing threads, reader: Traject::MarcReader and writer: Traject::SolrJsonWriter
2016-09-05T11:51:25-04:00 ERROR Error in Solr batch add. Will retry documents individually at performance penalty: Solr response: 404: <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 404 Not Found</title>
```

I am pretty sure that solr (5.3.1) is running. Can you guys help me out? Also, if I just want to visualize an existing core in Solr without any advance functions, how can I tell Blacklight the core name? I found this doc: https://github.com/projectblacklight/blacklight/wiki/Solr-Configuration Nothing mentioned about this though.

Thanks,

**Comment by mejackreed** [comment]
@Yongyao With regard to your question:

> Also, if I just want to visualize an existing core in Solr without any advance functions, how can I tell Blacklight the core name? 

You can start the Blacklight instance (Rails) and add an env variable to the location of that core

```sh
$ SOLR_URL="http://my.production.server/solr/core_name/" rails server
```

You may need to update some of your configurations (field names) in `app/controllers/catalog_controller.rb` to reference the field names for your core.

Checkout: https://github.com/projectblacklight/blacklight/wiki/Solr-Configuration#solr-configuration & https://github.com/projectblacklight/blacklight/wiki/Configuration---Facet-Fields

**Comment by barmintor** [comment]
Closing this ticket since the relevant issue to it (hydra-jetty vs solr_wrapper) is resolved, and the v7.0 quickstart issues are being tracked elsewhere.
